dts/bindings/i3c/i3c-device.yaml

```diff
@@ -0,0 +1,58 @@
+# Copyright (c) 2017, Linaro Limited
+# SPDX-License-Identifier: Apache-2.0
+
+# Common fields for I3C devices
+
+include: [base.yaml, power.yaml]
+
+on-bus: i3c
+
+properties:
+    reg:
+      required: true
+      description: |
+        Contains 3 fields.
+
+        For I3C devices, the 3 fields are static address, first half
+        of Provisioned ID, and the second half of the Provisioned ID.
+          1. The static address can be assigned to be the target device's
+             dynamic address before Dynamic Address Assignment (DAA)
+             starts. Can be zero if static address is not used, and
+             target address is determined via DAA.
+          2. First half of the Provisioned ID contains the manufacturer
+             ID left-shifted by 1, where the manufacturer ID is
+             the bits 33-47 (zero-based) of the 48-bit Provisioned ID.
+          3. Second half of the Provisioned ID contains the combination of
+             the part ID (bits 16-31 of the Provisioned ID) left-shifted
+             by 16, and the instance ID (bits 12-15 of the Provisioned ID)
+             left-shifted by 12. Basically, this is the lower 32 bits
+             of the Provisioned ID.
+
+        Note that the DT node of a I3C target device should be in format
+        "<node name>@<static address>", e.g. "sensor@42".
+
+        For I2C devices, the 3 fields are static address, 0x00, and
```

Comment:
yes, but for an existing driver such as `atmel,at24.yaml`. This would only exist for the i2c protocol and includes the `i2c-device.yaml`. If this is defined within the devicetree under an i3c bus... it will not bind and pull in it's definitions. That is what I was trying to fix with that pull request.
I don't think it's feasible to go into every i2c device driver and add the `DT_ON_BUS()` for i3c and have it load in the i2c api calls. It should just work with the existing i2c definition...